LLVM>=3.5 build failure

I am getting a build failure on arm on both llvm 3.6.1 and llvm 3.7.0.

````
    CC src/runtime_ccall.o
/home/viral/julia/src/runtime_ccall.cpp: In function ‘jl_value_t* jl_get_cpu_name()’:
/home/viral/julia/src/runtime_ccall.cpp:138:5: error: ‘StringRef’ was not declared in this scope
     StringRef HostCPUName = llvm::sys::getHostCPUName();
     ^
/home/viral/julia/src/runtime_ccall.cpp:138:5: note: suggested alternative:
In file included from /home/viral/julia/usr/include/llvm/ADT/StringMap.h:17:0,
                 from /home/viral/julia/usr/include/llvm/Support/Host.h:17,
                 from /home/viral/julia/src/runtime_ccall.cpp:131:
/home/viral/julia/usr/include/llvm/ADT/StringRef.h:40:9: note:   ‘llvm::StringRef’
   class StringRef {
         ^
/home/viral/julia/src/runtime_ccall.cpp:138:15: error: expected ‘;’ before ‘HostCPUName’
     StringRef HostCPUName = llvm::sys::getHostCPUName();
               ^
/home/viral/julia/src/runtime_ccall.cpp:140:31: error: ‘HostCPUName’ was not declared in this scope
     return jl_pchar_to_string(HostCPUName.data(), HostCPUName.size());
                               ^
/home/viral/julia/src/runtime_ccall.cpp:141:1: warning: control reaches end of non-void function [-Wreturn-type]
 }
 ^
make[1]: *** [runtime_ccall.o] Error 1
make: *** [julia-src-release] Error 2
````

--- a/src/intrinsics.cpp
+++ b/src/intrinsics.cpp
@@ -913,7 +913,7 @@ static jl_cgval_t emit_intrinsic(intrinsic f, jl_value_t **args, size_t nargs,
         f = fptosi_auto;
     unsigned expected_nargs = intrinsic_nargs[f];
     if (expected_nargs && expected_nargs != nargs) {
-        jl_errorf("intrinsic #%d %s: wrong number of arguments", f, jl_intrinsic_name((int)f));
+        jl_errorf("intrinsic #%d %s: wrong number of arguments", f, JL_I::jl_intrinsic_name((int)f));
     }
 
     switch (f) {

--- a/src/runtime_ccall.cpp
+++ b/src/runtime_ccall.cpp
@@ -3,6 +3,7 @@
 #include <map>
 #include <string>
 #include <cstdio>
+#include <llvm/Support/Host.h>
 #include "julia.h"
 #include "julia_internal.h"
 
@@ -128,7 +129,6 @@ void *jl_load_and_lookup(char *f_lib, char *f_name, uv_lib_t **hnd)
 }
 
 // miscellany
-#include <llvm/Support/Host.h>
 extern "C" DLLEXPORT
 jl_value_t *jl_get_cpu_name(void)
 {

--- a/src/runtime_ccall.cpp
+++ b/src/runtime_ccall.cpp
@@ -6,6 +6,7 @@
 #include <llvm/Support/Host.h>
 #include "julia.h"
 #include "julia_internal.h"
+using namespace llvm;
 
 // --- library symbol lookup ---
 

